R/spliceHelperFunctions.R

```diff
@@ -96,232 +96,180 @@ findSpliceOverlapsByDist <-function(query, subject, ignore.strand=FALSE, maxDist
   olap
 }
 
+#' annotate splice overlap by distance
+#' @noRd
 annotateSpliceOverlapsByDist <-function(query, subject) {
-
-  ## examples for test purposes
-  # query=rowRanges(seBambu.core)[c('ENST00000344579', # exon skipping, alternative TSS (-48), +, ENSG00000158109
-  #                                 'ENST00000270792', # intron retention subject 1(last exon), alt. TSS, alt. TES, +, ENSG00000
-  #                                 'ENST00000410032', # alternative first exon, exon skipping query: 2, exon skipping subject: 0, alternative TSS (2bp only), internalFirstExon.subject +
-  #                                 'ENST00000468178', # alternative last exon +
-  #                                 'ENST00000485956', # alternative first exon, alternative last exon, exon skipping subject = 1, internal first exon query, +
-  #                                 'ENST00000530807', # exon skipping query 1, alternative TSS (-17),  -
-  #                                 'ENST00000409894',  # alternative 3' exon splice site, exon skipping query 2, alternative TSS, alterantive TES, +, ENSG00000125630
-  #                                 'ENST00000524447',  # alternative TSS, alternative last exon (internal), alternative exon 3' end,-, ENSG00000165916
-  #                                 'ENST00000591696' # alternative TSS, alternative 3' exon (2), alternative 5' exon (1) alternative TES, ,+,ENSG00000141349
-  #                                 )]
-  # subject=rowRanges(seBambu.core)[c('ENST00000378344',
-  #                                   'ENST00000319041',
-  #                                   'ENST00000338530',
-  #                                   'ENST00000338530',
-  #                                   'ENST00000338530',
-  #                                   'ENST00000532718',
-  #                                   'ENST00000263331',
-  #                                   'ENST00000602866',
-  #                                   'ENST00000585361')]
-  # query <- rep(query,2000)
-  # subject <- rep(subject,2000)
-  # 
-
   queryStartRng <- ranges(selectStartExonsFromGrangesList(query, exonNumber = 1))
```

Comment:
is there a fast way to extract the first/last range from a ranges object with strand information (not granges)? If this can be extracted in a short amount of time, then we can extract this in the sub-functions every time this is required (this would simplify the number of arguments a lot, but this has to quite fast to avoid re-computing this if it is slow)

Reply:
i.e. something like selectStartExonFromRangesList(range, strand) + selectEndExonFromRangesList(range, strand)  which could be applied on subjectFullRng/queryFullRng